### PR TITLE
add DLM FSI example

### PR DESCRIPTION
**Pull request by boyceg**
### IBAMR Pull Request Checklist
- [x] Does the test suite pass?
- [x] Was clang-format run?
- [x] Were relevant issues cited? Please remember to add `Fixes #12345` to close
      the issue automatically if we are fixing the problem.
- [x] Is this a change others will want to know about? If so, then has a
      changelog entry been added?
- [x] If new data structures have been added to a class then have they been set
      up appropriately for restarts? If so, ensure that the restart version
      number is incremented.
- [x] Does this change include a bug fix or new functionality? If so, a new test
      or tests should be added. New tests should run quickly (less than a minute
      in release mode). If possible, an older test should gain a new option so
      that we do not need to compile more test executables.
- [x] Did you (if your account has permission to do so) set relevant labels on
      GitHub for the pull request?

### CODE REVIEW

**Comment by boyceg** [comment]
@jabrown893: this is an example of how to set up a model that uses a separate (explicit) mechanics solver with subcycling in the mechanics solver. We should try to do the same thing for your TAVR models --- at least for the stents, which will have a mass density larger than the background fluid.

**Comment by boyceg** [comment]
@amneetb or @Raamkrishnan, would one of you all be able to make a version of this new example that uses the multiphase flow solver instead of the constant coefficient solver?

This would be variable density and (for now) constant viscosity, with the density field _prescribed_ by the IB model. I.e., the density field should not be advected by the fluid solver.

File: examples/DLM/ex0/example.cpp
Comment (drwells) [inline examples/DLM/ex0/example.cpp:619]: Interesting approach. A major source of complication in `DLMMethod` (from the DLM project) was that we interpolated positions in time - it looks like that is skipped here.

Comment (boyceg) [reply examples/DLM/ex0/example.cpp:619]: You don't need to do that if you couple via fractional stepping.

Comment (boyceg) [reply examples/DLM/ex0/example.cpp:619]: I think fractional stepping (e.g. using Strang) makes a lot of sense here because the idea (at least for stiff elasticity) is that you want to be taking smaller structural time steps _anyway_.

Comment (drwells) [reply examples/DLM/ex0/example.cpp:619]: Is there any loss of accuracy from holding the IBFE position constant over the substeps?

Comment (boyceg) [reply examples/DLM/ex0/example.cpp:619]: Surely it impacts accuracy --- but I think it does not impact order of accuracy. Also, I don't think this is the major determinant of accuracy here.

(BTW, we do the same thing here: https://arxiv.org/abs/2204.12907.)

Comment (boyceg) [reply examples/DLM/ex0/example.cpp:619]: BTW, presumably the better way to do this is to use a more modern time stepping scheme that properly handles multiple time scales.

Comment (drwells) [reply examples/DLM/ex0/example.cpp:619]: I might give this a try with the deal.II code later :)

Comment (boyceg) [reply examples/DLM/ex0/example.cpp:619]: I just added some input files demonstrating how to increase structural stiffness without reducing the overall time step size (although, of course, it is necessary to increase the number of structural sub-steps).

**Comment by drwells** [comment]
We may be missing some input files. The CMake stuff should be OK once things compile cleanly - we can see the example being compiled in the log.

**Comment by boyceg** [comment]
I want to double check one of the numerical parameters before merging...

**Comment by amneetb** [comment]
> @amneetb or @Raamkrishnan, would one of you all be able to make a version of this new example that uses the multiphase flow solver instead of the constant coefficient solver?
> 
> This would be variable density and (for now) constant viscosity, with the density field _prescribed_ by the IB model. I.e., the density field should not be advected by the fluid solver.

Done ;-)

https://github.com/IBAMR/IBAMR/tree/amneetb/add-multiphase-elasticity-example/examples/multiphase_flow/ex14

**Comment by boyceg** [comment]
I think that this is ready to merge.

**Comment by boyceg** [comment]
@drwells, does this look OK?

**Review by drwells — APPROVED** [review]
I have some minor suggestions. I see some nice vortex shedding with `input2d`.

**Comment by boyceg** [comment]
Thanks for the updates. Will merge once the CI is happy.

**Pull request merged**
